src/tqec/compile/observables.py

```diff
@@ -144,8 +144,10 @@ def inplace_add_observables(
         for pipe in observable.bottom_regions:
             z = pipe.u.position.z
             template = template_slices[z]
+            origin = template.origin_shift
+            shifted_pipe = pipe.shift_position_by(-origin.x, -origin.y)
```

Comment:
I do not get why you need to shift the pipe now. What changed to require this shift?

Reply:
The `LayoutTemplate` has some shifts in the space and we need to account for it when instantiating the abstract observables. Actually this is a bug left behind. I did not notice that before because the logical cnot and some other tests cases I used have all the `LayoutTemplate`s with origin at `Position2D(0, 0)`. I found this problem when I have added more test cases in `compile_test.py`.